src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/IndexInstanceV6.sql

```diff
@@ -47,7 +47,7 @@ BEGIN
             @seriesKey = SeriesKey,
             @instanceKey = InstanceKey,
             @status = Status
-        FROM dbo.Instance WITH (HOLDLOCK)
+        FROM dbo.Instance WITH (UPDLOCK)
```

Comment:
We don't necessarily care if a new row is inserted during this time (as any new rows must be indexed using the newly added tags). So we can reduce the granularity to an `UPDLOCK`